types/v56/openapi.go

```diff
@@ -175,26 +176,38 @@ type VdcComputePolicy struct {
 		AdditionalProp2 string `json:"additionalProp2,omitempty"`
 		AdditionalProp3 string `json:"additionalProp3,omitempty"`
 	} `json:"extraConfigs,omitempty"`
-	PvdcComputePolicyRef *struct {
-		Name string `json:"name,omitempty"`
-		ID   string `json:"id,omitempty"`
-	} `json:"pvdcComputePolicyRef,omitempty"`
-	PvdcComputePolicy *struct {
-		Name string `json:"name,omitempty"`
-		ID   string `json:"id,omitempty"`
-	} `json:"pvdcComputePolicy,omitempty"`
-	CompatibleVdcTypes []string `json:"compatibleVdcTypes,omitempty"`
-	IsSizingOnly       bool     `json:"isSizingOnly,omitempty"`
-	PvdcID             string   `json:"pvdcId,omitempty"`
-	NamedVMGroups      [][]struct {
-		Name string `json:"name,omitempty"`
-		ID   string `json:"id,omitempty"`
-	} `json:"namedVmGroups,omitempty"`
-	LogicalVMGroupReferences []struct {
-		Name string `json:"name,omitempty"`
-		ID   string `json:"id,omitempty"`
-	} `json:"logicalVmGroupReferences,omitempty"`
-	IsAutoGenerated bool `json:"isAutoGenerated,omitempty"`
+	PvdcComputePolicyRef     *OpenApiReference   `json:"pvdcComputePolicyRef,omitempty"`
+	PvdcComputePolicy        *OpenApiReference   `json:"pvdcComputePolicy,omitempty"`
+	CompatibleVdcTypes       []string            `json:"compatibleVdcTypes,omitempty"`
+	IsSizingOnly             bool                `json:"isSizingOnly,omitempty"`
+	PvdcID                   string              `json:"pvdcId,omitempty"`
+	NamedVMGroups            []OpenApiReferences `json:"namedVmGroups,omitempty"`
+	LogicalVMGroupReferences OpenApiReferences   `json:"logicalVmGroupReferences,omitempty"`
+	IsAutoGenerated          bool                `json:"isAutoGenerated,omitempty"`
+}
+
+// VdcComputePolicyV2 contains VDC specific configuration for workloads (version 2.0.0)
+// https://developer.vmware.com/apis/vmware-cloud-director/latest/data-structures/VdcComputePolicy2/
+type VdcComputePolicyV2 struct {
```

Comment:
The idea here is that someone can still use V1 without being able to access V2 fields, while V2 can access everything (like VCD API does at the moment).